diff --git a/node_modules/xo/lib/options-manager.js b/node_modules/xo/lib/options-manager.js
index 6feb6ec..1d6db87 100644
--- a/node_modules/xo/lib/options-manager.js
+++ b/node_modules/xo/lib/options-manager.js
@@ -108,7 +108,11 @@ const mergeWithFileConfig = async options => {
 	}
 
 	const searchPath = options.filePath || options.cwd;
-	const {config: xoOptions, filepath: xoConfigPath} = (await configExplorer.search(searchPath)) || {};
+	let {config: xoOptions, filepath: xoConfigPath} = (await configExplorer.search(searchPath)) || {};
+  // wrong nested `package.json` resolved
+  if (!xoOptions && xoConfigPath) {
+    ({config: xoOptions, filepath: xoConfigPath} = (await configExplorer.search(path.resolve(xoConfigPath, '../..'))) || {});
+  }
 	const {config: enginesOptions} = (await pkgConfigExplorer.search(searchPath)) || {};
 
 	options = normalizeOptions({
